File: Javascript/Stack.js
class Node {
    constructor(data, node){
        this.data = data;
        this.link = node;
    }

    setLink = (node) => {
        this.link = node;
    }

    getLink = () => {
        return this.link;
    }
}

class Stack {
    constructor(){
        this.head = null;
    }

    push = (data) => {
        let temp = new Node(data, null)
        if (this.head === null){
            this.head = temp;
        } else {
            temp.link = this.head;
            this.head = temp;
        }
    }

    peek = () => {
        let temp = this.head;
        if (temp === null){
            console.log("Stack Empty!");
        }
        return temp.data;
    }

    pop = () => {
        let temp = this.head;
        if (temp === null){
            console.log("Stack Empty!");
        }
        let data = temp.data;
        this.head = temp.link;
        return data;
    }

    isEmpty = () => {
        if (this.head === null){
            return true;
        } else {
            return false;
        }
    }

    size = () => {
        let temp = this.head;
        let counter = 0;
        while (temp != null){
            counter += 1;
            temp = temp.link;
        }
        return counter;
    }

    printList = () => {
        let temp = this.head;
        while (temp != null){
            console.log(temp.data);
            temp = temp.link;
        }
    }
}

let stack = new Stack();

console.log(stack.isEmpty());

stack.push(1);
stack.push(3);
stack.push(5);
stack.push(7);
stack.push(9);

console.log(stack.isEmpty());

stack.printList();
console.log(stack.size());

console.log(stack.peek());
console.log(stack.pop());
console.log(stack.peek());
stack.printList();

File: Python/Stack.py
class Node:
    def __init__(self, data, node=None):
        self.data = data
        self.link = node
    
    def set_link(self, node):
        self.link = node

    def get_link(self):
        return self.link

class Stack:
    def __init__(self):
        self.head = None

    def push(self, data):
        temp = Node(data)
        if self.head == None:
            self.head = temp
        else:
            temp.link = self.head
            self.head = temp

    def pop(self, data):
        temp = self.head
        if temp == None:
            print("Stack Empty!")
        val = temp.data
        self.head = temp.link
        return val

    def peek(self):
        temp = self.head
        if temp == None:
            print("Stack Empty!")
        return temp.data 

    def isEmpty(self):
        if self.head == None:
            return True
        return False

    def print_list(self):
        temp = self.head
        while temp != None:
            print(temp.data)
            temp = temp.link

    def size(self):
        temp = self.head
        counter = 0
        while temp != None:
            temp = temp.link
            counter += 1
        return counter

def main():
    print("Hello, There!")
    
    stack = Stack()

    print(stack.isEmpty())
    stack.push(1)
    stack.push(3)
    stack.push(5)
    stack.push(7)
    stack.push(9)
    
    print(stack.peek())

    stack.print_list()
    print(stack.pop())
    print(stack.isEmpty())
    stack.print_list()


if __name__ == "__main__":
    main()
